crates/miden-objects/src/account/component/metadata/storage/schema.rs

```diff
@@ -0,0 +1,1202 @@
+use alloc::boxed::Box;
+use alloc::collections::BTreeMap;
+use alloc::string::{String, ToString};
+use alloc::vec::Vec;
+use core::iter;
+
+use miden_core::utils::{ByteReader, ByteWriter, Deserializable, Serializable};
+use miden_processor::DeserializationError;
+
+use super::type_registry::{InitValueRequirement, SCHEMA_TYPE_REGISTRY, SchemaTypeIdentifier};
+use super::{InitStorageData, StorageValueName, WordValue};
+use crate::account::{AccountStorage, StorageMap, StorageSlot, StorageSlotName};
+use crate::errors::AccountComponentTemplateError;
+use crate::{Felt, FieldElement, Word};
+
+/// Alias used for iterators that collect all init-required values and their types within a
+/// component storage schema.
+pub type SchemaRequirementsIter<'a> =
+    Box<dyn Iterator<Item = (StorageValueName, InitValueRequirement)> + 'a>;
+
+// STORAGE SCHEMA
+// ================================================================================================
+
+/// Describes the storage layout of an account component in terms of named storage slots.
+#[derive(Debug, Clone, Default, PartialEq, Eq)]
+pub struct AccountStorageSchema {
+    fields: BTreeMap<StorageSlotName, StorageSlotSchema>,
```

Comment:
nit: I know `fields` comes from my proposal - but I wonder if `slots` would be a better name here.